File: flight_choice_prediction/v2/post/Node SDK/flight_choice_prediction.js
const WAConvo = require("waconvo");

const waconvo = new WAConvo({
  clientId: "YOUR_API_KEY",
  clientSecret: "YOUR_API_SECRET",
});

async function main() {
  try {
    const flightOffersResponse = await waconvo.shopping.flightOffersSearch.get({
      originLocationCode: "SYD",
      destinationLocationCode: "BKK",
      departureDate: "2022-08-01",
      adults: "2",
    });
    const response = await waconvo.shopping.flightOffers.prediction.post(
      flightOffersResponse
    );
    console.log(response.data);
  } catch (error) {
    console.error(error);
  }
}

main();

File: flight_choice_prediction/v2/post/iOS SDK/FlightChoicePrediction.swift
import WAConvo

var waconvo: WAConvo = WAConvo(
    client_id: "YOUR_API_ID",
    client_secret: "YOUR_API_SECRET"
)


waconvo.shopping.flightOffersSearch.get(params:
   ["originLocationCode": "MAD",
    "destinationLocationCode": "BER",
    "departureDate": "2022-05-16",
    "returnDate": "2022-05-30",
    "adults": "2"],
    onCompletion: { result in

        switch result {
        case let .success(response):
            self.waconvo.shopping.flightOffers.prediction.post(body: response.result, onCompletion: {
                                                                   predictionResult in

                                                                   switch predictionResult {
                                                                   case let .success(response):
                                                                       print(response.data)
                                                                   case let .failure(error):
                                                                       fatalError(error.localizedDescription)
                                                                   }
                                                               })
        case let .failure(error):
            fatalError(error.localizedDescription)
        }
    })
